### PR TITLE
bus predictions type and resolvers

### PR DESCRIPTION
**Pull request by nathanchica**
This pull request introduces a new DataLoader for fetching bus stop predictions, removes unused fields from the bus stop prediction data structures, and adds comprehensive tests for both bus position and bus stop loaders. It also introduces a new `BusDirection` enum to the GraphQL schema and updates the context and mocks to support the new loader. These changes improve data fetching efficiency, streamline the prediction data model, and enhance test coverage.

**Bus Stop Predictions Loader and Context Integration:**
- Added a new `createBusStopPredictionsLoader` DataLoader in `busStop.ts` for efficient, batched fetching of bus stop predictions from the ACTRealtime service. This loader groups requests by route and stop, and is integrated into the app context, mocks, and loader types. [[1]](diffhunk://#diff-d5830bc8af2993269387d0196ae0b68f7bcc8b279780c53643c0a78eb062aeb9R23-R64) [[2]](diffhunk://#diff-0cd594dddd6a9f2e3e26afebbb92716434437d7001c2416252aa7531e7f2b8d4L6-R11) [[3]](diffhunk://#diff-0cd594dddd6a9f2e3e26afebbb92716434437d7001c2416252aa7531e7f2b8d4R22) [[4]](diffhunk://#diff-0cd594dddd6a9f2e3e26afebbb92716434437d7001c2416252aa7531e7f2b8d4R39) [[5]](diffhunk://#diff-c580f636f8b6be76f538d622282e23ebfc53df49ade99e12288a324db36210c8L5-R5) [[6]](diffhunk://#diff-c580f636f8b6be76f538d622282e23ebfc53df49ade99e12288a324db36210c8R59)

**Bus Stop Prediction Data Model Simplification:**
- Removed the `departureTime` and `distanceToStopFeet` fields from the `BusStopPrediction` type and associated formatter functions, simplifying the prediction structure. [[1]](diffhunk://#diff-ea7ed625ead0cf3a79928efdeff7f1f939e003dbf57fe91763c2ee65fb714546L10-L13) [[2]](diffhunk://#diff-ea7ed625ead0cf3a79928efdeff7f1f939e003dbf57fe91763c2ee65fb714546L57-L60) [[3]](diffhunk://#diff-ea7ed625ead0cf3a79928efdeff7f1f939e003dbf57fe91763c2ee65fb714546L97) [[4]](diffhunk://#diff-ea7ed625ead0cf3a79928efdeff7f1f939e003dbf57fe91763c2ee65fb714546L107-L109)
- Updated all relevant tests to reflect the new, simplified shape of bus stop prediction data. [[1]](diffhunk://#diff-b1535ca0e8498a646c340f7b88d6a811c515c78b1f9e8ca30015b7bddfe6a51dL58-L69) [[2]](diffhunk://#diff-b1535ca0e8498a646c340f7b88d6a811c515c78b1f9e8ca30015b7bddfe6a51dL88) [[3]](diffhunk://#diff-b1535ca0e8498a646c340f7b88d6a811c515c78b1f9e8ca30015b7bddfe6a51dL179-L189)

**Testing Improvements:**
- Added new test suites for `busPosition` and `busStop` loaders, covering batching, error handling, and edge cases for both profiles and predictions. [[1]](diffhunk://#diff-b2bc1d514cfbc299e03fa3013fdd2abf910e98142ad4bf169ec7bad8b7fc2a7fR1-R58) [[2]](diffhunk://#diff-3e4093a135fd72b36d41eaa9207bbedb9b7c77a10ed0d8116256f3677f1ce62cR1-R74)

**GraphQL Schema and Type Updates:**
- Introduced a `BusDirection` enum (`INBOUND`/`OUTBOUND`) to the GraphQL schema and corresponding TypeScript types, standardizing direction handling across the codebase. [[1]](diffhunk://#diff-ea303162bc6e66e40417747c613bdde8df2e4bb62d5684e599f70292fe7e40c3R2-R16) [[2]](diffhunk://#diff-40344d3b7617e813e06bc1eda980de43f5f81d0f6abcec874c40b7d94b16d397R8-R9)

**Test and Mock Context Enhancements:**
- Updated the mock context and test client setup to support the new predictions loader and direction enum, ensuring all new functionality is testable in isolation. [[1]](diffhunk://#diff-ead7cd87c430c6d657c9f39c92ae4f2e9ca9d79def837fcaff4cd547a1d3f341L3-R11) [[2]](diffhunk://#diff-c580f636f8b6be76f538d622282e23ebfc53df49ade99e12288a324db36210c8L5-R5) [[3]](diffhunk://#diff-c580f636f8b6be76f538d622282e23ebfc53df49ade99e12288a324db36210c8R59)

These changes collectively improve the maintainability, efficiency, and testability of bus stop prediction data handling in the codebase.

### CODE REVIEW

File: src/schema/busStop/busStop.resolver.ts [hunk 1]
```diff
@@ -45,4 +46,36 @@ export const busStopResolvers: Resolvers = {
...
+                const key = { routeId, stopCode, direction } as const;
+
+                const initialPredictions = await context.loaders.busStop.predictions.clear(key).load(key);
+                yield { busStopPredictions: initialPredictions };
+
+                const intervalMs = context.env.AC_TRANSIT_POLLING_INTERVAL;
+                while (true) {
+                    const predictions = await context.loaders.busStop.predictions.clear(key).load(key);
+                    yield { busStopPredictions: predictions };
+                    await new Promise((resolve) => setTimeout(resolve, intervalMs));
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/schema/busStop/busStop.resolver.ts:76]: gql yoga properly handles when subscriptions end or gets terminated

File: src/schema/busStop/busStop.resolver.ts [hunk 2]
```diff
@@ -45,4 +46,36 @@ export const busStopResolvers: Resolvers = {
...
+
+                const key = { routeId, stopCode, direction } as const;
+
+                const initialPredictions = await context.loaders.busStop.predictions.clear(key).load(key);
+                yield { busStopPredictions: initialPredictions };
+
+                const intervalMs = context.env.AC_TRANSIT_POLLING_INTERVAL;
+                while (true) {
+                    const predictions = await context.loaders.busStop.predictions.clear(key).load(key);
+                    yield { busStopPredictions: predictions };
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/schema/busStop/busStop.resolver.ts:75]: the service layer already has caching we don't want another layer of caching at dataloaders
